src/SessionTracker.php

```diff
@@ -281,27 +286,36 @@ protected function constructPayload($sessions)
     protected function deliverSessions()
     {
         $sessions = $this->getSessionCounts();
+
         $this->setSessionCounts([]);
-        if (count($sessions) == 0) {
+
+        if (!is_array($sessions) || count($sessions) === 0) {
             return;
         }
+
         if (!$this->config->shouldNotify()) {
             return;
         }
+
         $http = $this->config->getSessionClient();
-        $payload = $this->constructPayload($sessions);
-        $headers = [
-            'Bugsnag-Api-Key' => $this->config->getApiKey(),
-            'Bugsnag-Payload-Version' => self::$SESSION_PAYLOAD_VERSION,
-            'Bugsnag-Sent-At' => strftime('%Y-%m-%dT%H:%M:%S'),
+
+        $options = [
+            'json' => $this->constructPayload($sessions),
+            'headers' => [
+                'Bugsnag-Api-Key' => $this->config->getApiKey(),
+                'Bugsnag-Payload-Version' => self::$SESSION_PAYLOAD_VERSION,
+                'Bugsnag-Sent-At' => strftime('%Y-%m-%dT%H:%M:%S'),
+            ],
         ];
+
         $this->setLastSent();
 
         try {
-            $http->post('', [
-                'json' => $payload,
-                'headers' => $headers,
-            ]);
+            if (method_exists($http, 'request')) {
+                $http->request('POST', '', $options);
+            } else {
+                $http->post('', $options);
+            }
```

Comment:
This isn't ideal but the return type of `Configuration::getSessionClient` is `GuzzleHttp\ClientInterface`, which doesn't have a `post` method in version 6 (it's defined on the concrete class instead)

We can tidy this up by depending on the concrete `GuzzleHttp\Client` class instead, but we have similar checks to this elsewhere so I've not done that as part of this PR